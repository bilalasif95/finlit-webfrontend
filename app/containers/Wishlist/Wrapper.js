import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 1.5rem 0 0;
  .wishlist_cont {
    margin: 1.5rem 0 0;
    .wishlist_item {
      padding: 20px 20px;
      border: 1px solid #d8d8d8;
      border-radius: 0.313rem;
      margin-bottom: 20px;
      .wishlist_details {
        display: flex;
        align-items: center;
        justify-content: space-between;
        width: 100%;
        margin-bottom: 22px;
        .action_btns {
          position: absolute;
          right: 0;
          top: 0;
          button {
            background-color: transparent;
            border: none;
            font-size: 1.25rem;
            padding: 0;
            margin: 0 0 0 1rem;
            &:focus {
              box-shadow: none;
            }
            &:last-child {
              svg {
                color: #46a135;
              }
            }
          }
        }
        .wishlist_img {
          height: 92px;
          width: 11%;
          img {
            height: 100%;
            object-fit: cover;
            border-radius: 5px;
          }
        }
        .wishlist_desc {
          width: 89%;
          padding: 0 0 0 1rem;
          position: relative;
          h2 {
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            width: 230px;
          }
          h2,
          h3 {
            font-size: 20px;
            color: #404041;
            margin-bottom: 0px;
            line-height: 40px;
          }
          p {
            font-size: 14px;
            color: #8b8b8b;
          }
        }
      }
      .wishlist_outcomes {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 50%;
        margin-bottom: 0px;
        li {
          color: #404041;
          font-size: 14px;
          font-weight: 700;
          display: flex;
          align-items: center;
          svg {
            margin-right: 10px;
            color: #8b8b8b;
            width: 20px;
            height: 20px;
          }
          img {
            margin-right: 10px;
            width: 20px;
            height: 20px;
          }
        }
      }
    }
    @media (max-width: 991px) {
      .wishlist_item {
        .wishlist_outcomes {
          width: 75%;
        }
      }
    }
    @media (max-width: 767px) {
      .wishlist_item {
        .wishlist_details {
          .wishlist_img {
            width: 15%;
          }
          .wishlist_desc {
            width: 85%;
          }
        }
        .wishlist_outcomes {
          width: 100%;
        }
      }
    }
    @media (max-width: 575px) {
      .wishlist_item {
        .wishlist_details {
          .wishlist_img {
            width: 20%;
          }
          .wishlist_desc {
            width: 80%;
          }
        }
        .wishlist_outcomes {
          flex-wrap: wrap;
          width: 100%;
        }
      }
    }
    @media (max-width: 480px) {
      .wishlist_item {
        padding: 0.75rem;
        .wishlist_details {
          .wishlist_img {
            width: 20%;
          }
          .wishlist_desc {
            width: 80%;
            h2,
            h3 {
              font-size: 16px;
            }
          }
        }
        .wishlist_outcomes {
          flex-wrap: wrap;
          justify-content: flex-start;
          width: 100%;
          li {
            font-size: 12px;
            width: 50%;
          }
        }
      }
    }
    @media (max-width: 414px) {
      .wishlist_item {
        padding: 0.75rem;
        .wishlist_details {
          flex-wrap: wrap;
          .wishlist_img {
            height: 100%;
            width: 100%;
            img {
              width: 100%;
            }
          }
          .wishlist_desc {
            padding: 0;
            width: 100%;
          }
        }
      }
    }
  }
`;

export default Wrapper;
