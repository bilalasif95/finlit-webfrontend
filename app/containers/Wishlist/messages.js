/*
 * Wishlist Page Messages
 *
 * This contains all the text for the Wishlist Page.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.WishlistPage';

export default defineMessages({
  ChangeProfile: {
    id: `${scope}.ChangeProfile`,
    defaultMessage: 'Upload Photo',
  },
  Tagline: {
    id: `${scope}.Tagline`,
    defaultMessage: `Complete Programming with Python : HandsOn Introduction for Beginners`,
  },
  ChangePassword: {
    id: `${scope}.ChangePassword`,
    defaultMessage: 'Change Password',
  },
  Desc: {
    id: `${scope}.Desc`,
    defaultMessage: `Aenean sed lorem est. Sed quis neque ut nibh suscipit
    imperdiet ac non augue. Aenean ornare sit amet lectus
    non tristique. Nunc ut volutpat lectus. Nulla 
    velit augue, pulvinar sed nisi sit amet, eleifend fermentum est.`,
  },
});
