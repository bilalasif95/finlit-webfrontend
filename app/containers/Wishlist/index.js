/*
 * Wishlist Page
 *
 */
import React from 'react';
// import InputBase from '@material-ui/core/InputBase';
import { Helmet } from 'react-helmet';
// import { FormattedMessage } from 'react-intl';
import '../../components/student-panel/Header/profile.css';
// import { withStyles } from '@material-ui/core/styles';
import {
  Container,
  Row,
  Col,
  Button,
  Breadcrumb,
  BreadcrumbItem,
} from 'reactstrap';
import { Link } from 'react-router-dom';
import {
  AiFillStar,
  AiOutlineClockCircle,
  // AiOutlineDelete,
} from 'react-icons/ai';
import { BiUser } from 'react-icons/bi';
import { GrCart } from 'react-icons/gr';
import { FaHeart } from 'react-icons/fa';
// import messages from './messages';
import Wrapper from './Wrapper';
import CartImg from '../../images/cartImg.png';
import CartImg1 from '../../images/cartImg1.png';
import Bag from '../../images/bag.png';

export default function Wishlist() {
  return (
    <div className="sub_pages">
      <Helmet>
        <title>Wishlist</title>
        <meta name="description" content="Wishlist Page" />
      </Helmet>
      <Wrapper>
        <Container fluid="xl">
          <Row>
            <Col lg={12}>
              <div className="wishlist">
                <div className="breadcrumb_bar">
                  <Breadcrumb>
                    <BreadcrumbItem>
                      <Link to="/">Home</Link>
                    </BreadcrumbItem>
                    <BreadcrumbItem active>Wishlist</BreadcrumbItem>
                  </Breadcrumb>
                </div>
                <div className="wishlist_cont">
                  <div className="wishlist_item">
                    <div className="wishlist_details">
                      <div className="wishlist_img">
                        <img className="img-fluid" src={CartImg} alt="Cart" />
                      </div>
                      <div className="wishlist_desc">
                        <h2>Programming with Python</h2>
                        <p>By Kirill Eremenko</p>
                        <h3>$44.99</h3>
                        <div className="action_btns">
                          <Button>
                            <GrCart />
                          </Button>
                          <Button>
                            <FaHeart />
                          </Button>
                        </div>
                      </div>
                    </div>
                    <ul className="list-unstyled wishlist_outcomes">
                      <li>
                        <AiFillStar />
                        Course
                      </li>
                      <li>
                        <BiUser />
                        1917 Students
                      </li>
                      <li>
                        <AiOutlineClockCircle />
                        44.5Total hours
                      </li>
                      <li>
                        <img src={Bag} alt="Bag" />
                        All Levels
                      </li>
                    </ul>
                  </div>
                  <div className="wishlist_item">
                    <div className="wishlist_details">
                      <div className="wishlist_img">
                        <img className="img-fluid" src={CartImg1} alt="Cart1" />
                      </div>
                      <div className="wishlist_desc">
                        <h2>Programming with Python</h2>
                        <p>By Kirill Eremenko</p>
                        <h3>$44.99</h3>
                        <div className="action_btns">
                          <Button>
                            <GrCart />
                          </Button>
                          <Button>
                            <FaHeart />
                          </Button>
                        </div>
                      </div>
                    </div>
                    <ul className="list-unstyled wishlist_outcomes">
                      <li>
                        <AiFillStar />
                        Course
                      </li>
                      <li>
                        <BiUser />
                        1917 Students
                      </li>
                      <li>
                        <AiOutlineClockCircle />
                        44.5Total hours
                      </li>
                      <li>
                        <img src={Bag} alt="Bag" />
                        All Levels
                      </li>
                    </ul>
                  </div>
                  <div className="wishlist_item">
                    <div className="wishlist_details">
                      <div className="wishlist_img">
                        <img className="img-fluid" src={CartImg} alt="Cart" />
                      </div>
                      <div className="wishlist_desc">
                        <h2>Programming with Python</h2>
                        <p>By Kirill Eremenko</p>
                        <h3>$44.99</h3>
                        <div className="action_btns">
                          <Button>
                            <GrCart />
                          </Button>
                          <Button>
                            <FaHeart />
                          </Button>
                        </div>
                      </div>
                    </div>
                    <ul className="list-unstyled wishlist_outcomes">
                      <li>
                        <AiFillStar />
                        Course
                      </li>
                      <li>
                        <BiUser />
                        1917 Students
                      </li>
                      <li>
                        <AiOutlineClockCircle />
                        44.5 Total hours
                      </li>
                      <li>
                        <img src={Bag} alt="Bag" />
                        All Levels
                      </li>
                    </ul>
                  </div>
                  <div className="wishlist_item">
                    <div className="wishlist_details">
                      <div className="wishlist_img">
                        <img className="img-fluid" src={CartImg1} alt="Cart1" />
                      </div>
                      <div className="wishlist_desc">
                        <h2>Programming with Python</h2>
                        <p>By Kirill Eremenko</p>
                        <h3>$44.99</h3>
                        <div className="action_btns">
                          <Button>
                            <GrCart />
                          </Button>
                          <Button>
                            <FaHeart />
                          </Button>
                        </div>
                      </div>
                    </div>
                    <ul className="list-unstyled wishlist_outcomes">
                      <li>
                        <AiFillStar />
                        Course
                      </li>
                      <li>
                        <BiUser />
                        1917 Students
                      </li>
                      <li>
                        <AiOutlineClockCircle />
                        44.5 Total hours
                      </li>
                      <li>
                        <img src={Bag} alt="Bag" />
                        All Levels
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </Col>
          </Row>
        </Container>
      </Wrapper>
    </div>
  );
}
