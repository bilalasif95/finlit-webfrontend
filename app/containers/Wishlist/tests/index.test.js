import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Wishlist from '../index';

describe('<Wishlist />', () => {
  it('should render its heading', () => {
    const { container } = render(
      <IntlProvider locale="en">
        <Wishlist />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
