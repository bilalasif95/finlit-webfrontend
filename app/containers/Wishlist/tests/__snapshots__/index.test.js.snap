// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutCourse /> should render its heading 1`] = `
<div
  class="sub_pages"
>
  <div
    class="Wrapper-sc-16iq352-0 cRHkYe"
  >
    <div
      class="container-xl"
    >
      <div
        class="about_course"
      >
        <div
          class="top_info"
        >
          <h6>
            About this course
          </h6>
          <b>
            <span>
              Complete Programming with Python : HandsOn Introduction for Beginners
            </span>
          </b>
          <p>
            Approved by 
            <b>
              FinLit
            </b>
          </p>
        </div>
        <div
          class="user_details"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-12 col-md-6 col-lg-4"
            >
              <ul>
                <li>
                  <p>
                    Student name
                  </p>
                  <p>
                    Chad Sanders
                  </p>
                </li>
                <li>
                  <p>
                    Language
                  </p>
                  <p>
                    English
                  </p>
                </li>
                <li>
                  <p>
                    Skill Level
                  </p>
                  <p>
                    Medium
                  </p>
                </li>
                <li>
                  <p>
                    Completion date
                  </p>
                  <p>
                    12/02/1978
                  </p>
                </li>
              </ul>
            </div>
            <div
              class="col-sm-12 col-md-6 col-lg-8"
            >
              <ul>
                <li>
                  <p>
                    Instructor
                  </p>
                  <p>
                    Beatrice Ross
                  </p>
                </li>
                <li>
                  <p>
                    Grade
                  </p>
                  <p>
                    A+
                  </p>
                </li>
                <li>
                  <p>
                    Marks
                  </p>
                  <p>
                    90%
                  </p>
                </li>
                <li>
                  <p>
                    Duration of course
                  </p>
                  <p>
                    2hr 30m
                  </p>
                </li>
                <li>
                  <p>
                    Description
                  </p>
                  <p>
                    <span>
                      Aenean sed lorem est. Sed quis neque ut nibh suscipit
    imperdiet ac non augue. Aenean ornare sit amet lectus
    non tristique. Nunc ut volutpat lectus. Nulla 
    velit augue, pulvinar sed nisi sit amet, eleifend fermentum est.
                    </span>
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="get_certificate"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-sm-3 col-md-3 col-lg-2"
            >
              <p>
                Certificates
              </p>
            </div>
            <div
              class="col-12 col-sm-9 col-md-9 col-lg-10"
            >
              <p>
                Get FinLit certificate by completing entire course
              </p>
              <button
                class="btn btn-secondary"
              >
                 FinLit certificate 
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
